Improved JS code architecture in the extension

valgreens opened the pull request:
Improved JS code architecture in the extension and also added a random color generator for each project. Removed jQuery dependency.

agonzalezro commented on README.md:
```diff
@@ -26,11 +26,10 @@ to:
 - Allow configuration for the "seen" projects background color.
 - Allow configuration for the list of colors.
-- Calculate the color instead set a random one?
+- Improve color load for projects when clicking on 'more' button
```
Minor comment: `.` at the end.

agonzalezro commented on colorify.js:
```diff
@@ -1,53 +1,63 @@
+        document.querySelector('.js-events-pagination').onclick = function () {
+          setTimeout(function () {
+            self.paint();
+          }, 1000);
```
Is this going to refresh every second? Why you didn't like the `click` event on the "Show more" button?

agonzalezro replied:
My bad. I was reading this as a `setInterval` and it didn't make a lot of sense.

agonzalezro commented on colorify.js:
```diff
@@ -1,53 +1,63 @@
+    [].forEach.call(document.querySelectorAll('.alert'), function (element) {
+      var project = self.getProjectName(element),
+          color = self.getColorForProject(project, previousColor);
+      console.log(color, project);
```
remove

agonzalezro commented on colorify.js:
```diff
@@ -1,53 +1,63 @@
+    var self = this,
+        previousColor;
+
+    [].forEach.call(document.querySelectorAll('.alert'), function (element) {
```
Is this right? I suspect that it's some weird JS trick :D

agonzalezro commented on colorify.js:
```diff
@@ -1,53 +1,63 @@
+
+  colorify.getColorForProject = function (project, previousColor) {
+    if (!(project in this.colors)) {
+      var color = ('00000' + (Math.random() * (1 << 24) | 0).toString(16)).slice(-6);
```
I would move this to a function with a meaningfully name

agonzalezro commented on colorify.js:
```diff
@@ -1,53 +1,63 @@
+  };
+
+  colorify.init();
+
```
Extra line?

agonzalezro commented on manifest.json:
```diff
@@ -15,7 +15,7 @@
     {
       "matches": ["https://github.com/*"],
-      "js": ["jquery.min.js", "colorify.js"]
+      "js": ["colorify.js"]
```
Awesome! :) Thanks!

valgreens commented:
I've made some changes: removed `console.log` statement, improved function names for readability and rewrote `popup.js` to work without jQuery. I think it's ready to merge when you have some time to check.

agonzalezro commented:
Nice one! I really like the changes, and the style is much better now as we discussed.

I continue not being sure that the distinction between projects is clear, but let's start with something!
